spec/features/admin/invoices/show_spec.rb

```diff
@@ -4,12 +4,92 @@
-
+    item = create(:item_with_invoices, invoices: [invoice], invoice_item_unit_price: 13000)
     visit "/admin/invoices/#{invoice.id}"
 
     expect(page).to have_content(invoice.id)
     expect(page).to have_content(invoice.status)
     expect(page).to have_content("Thursday, January 06, 2022")
     expect(page).to have_content(invoice.customer_name)
   end
+
+  it 'shows the invoiced items name' do
+    merchant = create(:merchant)
+    invoice = create(:invoice)
+    item = create(:item_with_invoices, merchant: merchant, invoices: [invoice], name: 'Toy')
+    item2 = create(:item_with_invoices, merchant: merchant, invoices: [invoice], name: 'Car')
+
+    visit "/admin/invoices/#{invoice.id}"
+
+    within("#invoice_#{invoice.invoice_items.first.id}") do
+      expect(page).to have_content("Item: Toy")
+    end
+
+    within("#invoice_#{invoice.invoice_items.second.id}") do
+      expect(page).to have_content("Item: Car")
+    end
+  end
+
+  it 'shows the quantity of the item ordered' do
+    merchant = create(:merchant)
+    invoice = create(:invoice)
+    item = create(:item_with_invoices, merchant: merchant, invoices: [invoice], invoice_quantity: 12)
+    item2 = create(:item_with_invoices, merchant: merchant, invoices: [invoice], invoice_quantity: 9)
+
+    visit "/admin/invoices/#{invoice.id}"
+
+    within("#invoice_#{invoice.invoice_items.first.id}") do
+      expect(page).to have_content("Quantity Ordered: 12")
+    end
+
+    within("#invoice_#{invoice.invoice_items.second.id}") do
+      expect(page).to have_content("Quantity Ordered: 9")
+    end
+  end
+
+  it 'shows the price the item sold for' do
+    merchant = create(:merchant)
+    invoice = create(:invoice)
+    item = create(:item_with_invoices, merchant: merchant, invoices: [invoice], invoice_item_unit_price: 13000)
+    item2 = create(:item_with_invoices, merchant: merchant, invoices: [invoice], invoice_item_unit_price: 24000)
+
+    visit "/admin/invoices/#{invoice.id}"
+
+    within("#invoice_#{invoice.invoice_items.first.id}") do
+      expect(page).to have_content("Unit Price: $130.00")
+    end
+
+    within("#invoice_#{invoice.invoice_items.second.id}") do
+      expect(page).to have_content("Unit Price: $240.00")
+    end
+  end
+
+  it 'shows the invoice items status' do
+    merchant = create(:merchant)
+    invoice = create(:invoice)
+    item = create(:item_with_invoices, merchant: merchant, invoices: [invoice], invoice_item_status: "shipped")
+    item2 = create(:item_with_invoices, merchant: merchant, invoices: [invoice], invoice_item_status: "pending")
+
+    visit "/admin/invoices/#{invoice.id}"
+
+    within("#invoice_#{invoice.invoice_items.first.id}") do
+      expect(page).to have_content("Status: shipped")
+    end
+
+    within("#invoice_#{invoice.invoice_items.second.id}") do
+      expect(page).to have_content("Status: pending")
+    end
+  end
+
+  it 'calculates the potential revenue of the invoice' do
```

Comment:
we should add more test data to check for invoice_items that are not associated with successful transactions.